src/seg_utils.py

```diff
@@ -129,171 +153,181 @@ def train_segmentation(
+
+            # compute validation step
+            if validationloader is not None:
+                with torch.no_grad():
+                    for x, y in validationloader:
+                        x = x.type(torch.FloatTensor)
                         y = y.type(torch.LongTensor)
-                        y = y.to(device).squeeze(1)
-
-                    # forward pass, compute validation loss and accuracy
-                    if use_amp is False:
-                        yhat = net(x)
-                        val_loss = criterion(yhat, y)
-                    else:
-                        with torch.cuda.amp.autocast():
+                        x = x.to(device)
+                        y = y.to(device)
+                        N_val = y.shape[0]
+                        tot_val += N_val
+                        if criterion.__class__.__name__ == 'CrossEntropyLoss':
+                            y = y.type(torch.LongTensor)
+                            y = y.to(device).squeeze(1)
+
+                        # forward pass, compute validation loss and accuracy
+                        if use_amp is False:
                             yhat = net(x)
                             val_loss = criterion(yhat, y)
+                        else:
+                            with torch.cuda.amp.autocast():
+                                yhat = net(x)
+                                val_loss = criterion(yhat, y)
+
+                        tmp_micro, tmp_macro = segmentation_metrics(yhat, y)
+                        running_F1_validation_micro += tmp_micro.item()
+                        running_F1_validation_macro += tmp_macro.item()
+
+                        # update running validation loss and accuracy
+                        running_validation_loss += val_loss.item()
+
+            loss = running_train_loss / len(trainloader)
+            F1_micro = running_F1_train_micro / len(trainloader)
+            F1_macro = running_F1_train_macro / len(trainloader)
+            train_loss.append(loss)
+            F1_train_trace_micro.append(F1_micro)
+            F1_train_trace_macro.append(F1_macro)
+
+            if validationloader is not None:
+                val_loss = running_validation_loss / len(validationloader)
+                F1_val_micro = running_F1_validation_micro / len(validationloader)
+                F1_val_macro = running_F1_validation_macro / len(validationloader)
+                validation_loss.append(val_loss)
+                F1_validation_trace_micro.append(F1_val_micro)
+                F1_validation_trace_macro.append(F1_val_macro)
+
+                live.log_metric("train/loss", loss)
+                live.log_metric("train/F1_micro", F1_micro)
+                live.log_metric("train/F1_macro", F1_macro)
+                live.log_metric("val/loss", val_loss)
+                live.log_metric("val/F1_micro", F1_val_micro)
+                live.log_metric("val/F1_macro", F1_val_macro)
+                live.next_step()
```

Comment:
I believe this will be discussed and covered in the other PR.

Reply:
> Is the DVC report (html) updated every epoch? I think we can use `live.make_report()` for this, so that the loss plot is updated live. Maybe this belongs to #11 @xiaoyachong

Yes, html is updated every epoch. 'live.next_step()' will automatically call 'live.make_report()'.